### PR TITLE
feat: clarify error messages around durable objects beta

### PR DESCRIPTION
**Pull request by nilslice**
Disambiguates error message described in #1859, adds more clarity around other places where Durable Object usage in beta may conflict with `wrangler` functionality.

### CODE REVIEW

File: src/deploy/schedule.rs [hunk 1]
```diff
@@ -37,12 +37,14 @@ impl ScheduleTarget {
...
-        if !res.status().is_success() {
-            anyhow::bail!(
-                "Something went wrong! Status: {}, Details {}",
-                res.status(),
-                res.text()?
-            )
+        let status = res.status();
+        if !status.is_success() {
+            let text = res.text()?;
+            if text.contains("workers.api.error.not_entitled") {
```
Comment (koeninger) [inline src/deploy/schedule.rs:43]: Is durable objects the only possible reason for not_entitled?

Comment (nilslice) [reply src/deploy/schedule.rs:43]: Probably not, but it's a temp quick fix until you get DO's to GA.

Comment (koeninger) [reply src/deploy/schedule.rs:43]: Then is there a reason not to show the original error plus a "If you are trying to use Modules, you need to opt in..."?

As it stands someone's probably going to be more confused, not less, if it has nothing to do with modules or DO

Comment (nilslice) [reply src/deploy/schedule.rs:43]: I'd think there would be a better API error, actually indicating the accurate entitlement missing. But I'm assuming you're not ready to commit to that until GA..

The original error was "Something went wrong!" + the ambiguous `workers.api.error.not_entitled`.

File: src/util.rs [hunk 2]
```diff
@@ -8,3 +8,5 @@ impl<'a, 'b> ApplyToApp<'a, 'b> for App<'a, 'b> {
         f(self)
     }
 }
+
+pub const TEMP_NOTICE_ES_MODULES_DO_BETA: &str = "While Durable Objects are in Beta, ES Modules are limited to accounts which have opted-in to the Beta. You may do so by following the instructions here: https://developers.cloudflare.com/workers/learning/using-durable-objects";
```
Comment (Electroid) [inline src/util.rs:12]:
```suggestion
pub const TEMP_NOTICE_ES_MODULES_DO_BETA: &str = "Your account does not have permission to do this! While Durable Objects are in Beta, the modules format is limited to accounts which have opted-in to the Beta. You may do so by following the instructions here: https://developers.cloudflare.com/workers/learning/using-durable-objects";
```

**Review by Electroid — CHANGES_REQUESTED** [review]

**Review by Electroid — APPROVED** [review]

**Pull request merged**